### ISSUE TITLE
[1.20.1 Quilt] Vertical Slabs don't render when using Quilted Fabric API

### ISSUE THREAD

**Issue by aninusmuffin** [Andrew6rant/Auto-Slabs#11]
Using AutoSlabs 1.1.0

### With Fabric API
https://mclo.gs/we5LXDv
![Screenshot from 2023-08-18 15-54-31](https://github.com/Andrew6rant/Auto-Slabs/assets/68734661/7ce5a09d-ec6a-419f-ae2f-4ca12910013b)

###  With Quilted Fabric API
https://mclo.gs/mhoxqm1
![Screenshot from 2023-08-18 15-56-21](https://github.com/Andrew6rant/Auto-Slabs/assets/68734661/03df8593-e618-4c87-8435-01b132300f17)

**Comment by Andrew6rant** [comment]
D'oh! This is almost certainly my fault for not testing Quilt Loader on every single release.

This is probably the offending line:
 https://github.com/Devan-Kerman/ARRP/blob/e12a601b9ef016aaacb17c03a2763ed6231612f2/src/main/java/net/devtech/arrp/mixin/LifecycledResourceManagerImplMixin.java#L37

I just assumed that Quilt would name its mod resources the same as Fabric for better compatibility, but that looks like it was a bad assumption to make.

Just to check as well, do the vertical slabs models in the Distinct Slabs embedded resource pack (Cut Sandstone, Polished Andesite, etc) work? They should be injected in a different way than my generated resources but I see warnings related to them in your logs.

**Comment by aninusmuffin** [comment]
> Just to check as well, do the vertical slabs models in the Distinct Slabs embedded resource pack (Cut Sandstone, Polished Andesite, etc) work? They should be injected in a different way than my generated resources but I see warnings related to them in your logs.

They also don't work :/
![image](https://github.com/Andrew6rant/Auto-Slabs/assets/68734661/a8ded314-0476-4369-80cd-55bd2ac89791)

**Comment by sylv256** [comment]
> 

So, is this a QFAPI bug, or is this your fault? It seems like something on QFAPI's side and you should probably report it. I don't think fixing their mistake is reasonable.

**Comment by DiosDonutSlave** [comment]
Can confirm this issue is on quilted fabrics side, as I have the issue with QFAPI as well

**Comment by corentin-godefroy** [comment]
i've the same problem... BUT without QFAPI.... -_- if anyone have a solution...

**Comment by Hand-Lock** [comment]
Having the same problem using QFAPI

**Comment by wzieciak** [comment]
I recreated this problem on Minecraft 1.20.1 with Quilt 0.22 and newest QFAPI package for it.

I see the block placement information is correctly set, but problem is with resources registration.
When I debug resources: [rrp.debug.zip](https://github.com/Andrew6rant/Auto-Slabs/files/13919451/rrp.debug.zip)
And make "resourcepack" from them: [autoslab.zip](https://github.com/Andrew6rant/Auto-Slabs/files/13919452/autoslab.zip)

Slabs are correctly displayed..

Maybe this is kind of typehint @Andrew6rant?

**Comment by wzieciak** [comment]
Here is default with resource pack provider by mod:
<img width="841" alt="Zrzut ekranu 2024-01-12 o 15 18 42" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/e837d867-a206-44ad-b8f0-9fcd65d893b3">
<img width="836" alt="Zrzut ekranu 2024-01-12 o 15 18 57" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/c6d75d63-5e7b-487e-9514-53cf6407b9ae">

Here is with added manually created resource pack on top:
<img width="836" alt="Zrzut ekranu 2024-01-12 o 15 19 38" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/b3f3d853-7bb8-4060-91ef-947eabba15a2">
<img width="847" alt="Zrzut ekranu 2024-01-12 o 15 19 50" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/86a854e0-1efb-4043-b451-f46147defce2">

When manually created resource pack is between:
<img width="846" alt="Zrzut ekranu 2024-01-12 o 15 20 44" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/11c69b72-7868-43ac-93f8-9cb4f161b85f">
<img width="849" alt="Zrzut ekranu 2024-01-12 o 15 21 09" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/695032a9-fce6-481d-a62c-1d1ed5283fd1">

When manually created resource pack is on bottom:
<img width="840" alt="Zrzut ekranu 2024-01-12 o 15 21 58" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/60730b16-d4df-4e35-b632-ddf969e5208d">
<img width="821" alt="Zrzut ekranu 2024-01-12 o 15 22 06" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/5d53f0d0-85da-4a89-8b8c-6a40e389fa8d">

So there is some problem on resource pack building process or other related process

**Comment by wzieciak** [comment]
And i got an idea :) 

Fabric API give us a "fabric" resource pack:
<img width="845" alt="Zrzut ekranu 2024-01-12 o 16 00 06" src="https://github.com/Andrew6rant/Auto-Slabs/assets/21332882/19665bf2-91cf-481b-bce6-b49d50c1765d">

This raw MC only with fabric api.


But Quilt and it's api doesn't 🙁

**Comment by wzieciak** [comment]
@Andrew6rant maybe for quilt it's the best to load your resource on the top of everything instead of using your custom implementation of ARRP injection method?

**Comment by wzieciak** [comment]
@Andrew6rant do you have any plans / ETA for fix that behaviour?

**Comment by RyanIsAinmDom** [comment]
@Andrew6rant
 I know it's kind of a long shot, but is there any chance of this issue getting fixed at this point? Such a shame, many mods I use are quilt-only and this is by far the best vertical slab mod.

**Comment by Andrew6rant** [comment]
Yeah, I haven't had much time/ motivation of late to work on Minecraft modding.

I am planning on fixing this issue (as well as a bunch of others) in the near future. I also welcome PRs.

**Comment by sylv256** [comment]
> @Andrew6rant I know it's kind of a long shot, but is there any chance of this issue getting fixed at this point? Such a shame, many mods I use are quilt-only and this is by far the best vertical slab mod.

kind of surprised that there are still quilt-only mods
